### ISSUE TITLE
Fix `custom-property-no-missing-var-function` false positives for style query in `if()` function

### ISSUE THREAD

**Issue by otomad** [stylelint/stylelint#8805]
### What minimal example or steps are needed to reproduce the bug?

```css
button {
	--state: normal;
	background-color: if(
		style(--state: hover): gray;
		else: white;
	);
}
```

### What minimal configuration is needed to reproduce the bug?

```json
{
  "rules": {
    "custom-property-no-missing-var-function": true
  }
}
```

### How did you run Stylelint?

[Demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcIBGBXALmiA7ABMAOlvmgLQkDOaAhmjPDlhAE4C2VANgNyHFJVgBrAOZMIKLFBKR2zegEsAZgAoeaYpQCe7GErKUadHAAsIANxhMAlPRFUN3ImtrtyhgO5G5tB8UsOAvoQgADQgCnLaAHJULHCIMAAeMQAO2gB0YOTkIeDY4UIIIAS4OPggTCja5GX0xTj1peAolBAsJMmiyRZoGiSMJCxyWXJYQiSmVEwkCuJgaHLYNThoFTCE9YFY-jmQWPkAYsxsaIUAVuTYObDJ2Yh1jZra7CNqCI3sBpRlwesPPU8vKR5ORjfQSSZQJZlD60L4gQjbfxAA)

### Which Stylelint-related dependencies are you using?

```json
{
  "stylelint": "16.25.0"
}
```

### What did you expect to happen?

We don't need `var()` function to enclose the custom property, where in the style query condition (`style()`) of `if()` function. `custom-property-no-missing-var-function` should not report the error.

### What actually happened?

```
4:9-16 error Unexpected missing var function for "--state" (custom-property-no-missing-var-function)
```

### Do you have a proposal to fix the bug?

Note that `if()` function will also raise `declaration-property-value-no-unknown` error, for temporarily solution please see #8779 . Since there are too many new CSS features raise `declaration-property-value-no-unknown` errors, I don't think it's worth creating issues for each of them.

Due to this issue is not related to upstream (MDN, CSSTree), I think this issue can be fixed manually by PR. However after fixing, it might solve `custom-property-no-missing-var-function` error only, you will still get the `declaration-property-value-no-unknown` error.

**Comment by sajdakabir** [comment]
Submitted PR #8813 to address this issue
